src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/SetAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,173 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+using Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System;
+
+namespace Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Cmdlet
+{
+    /// <summary>
+    /// Cmdlet to create a new Azure Sql Database Instance Failover Group
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlDatabaseInstanceFailoverGroup",
+        ConfirmImpact = ConfirmImpact.Medium)]
+    public class SetAzureSqlInstanceFailoverGroup : AzureSqlInstanceFailoverGroupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for set with an Input Object.
+        /// </summary>
+        protected const string SetIFGByInputObjectParameterSet =
+            "Set a Instance Failover Group from AzureSqlInstanceFailoverGroupModel instance definition";
+
+        /// <summary>
+        /// Parameter set name for set with a resource ID.
+        /// </summary>
+        private const string SetIFGByResourceIdSet = "Set a Instance Failover Group from Resource Id";
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@jugeorge same comment about removing `ValueFromPipelineByPropertyName` from the below three parameters and moving them into a separate parameter set from `InputObject` and `ResourceId`